float32 x
float32 y
float32 w
float32 h
---
float32 center_x
float32 center_y
